scripts/tt-console/console.c

```diff
@@ -697,6 +697,8 @@ static int loop(struct console *const cons)
 				putchar('\r');
 			}
 			(void)putchar(ch);
+			/* Flush to STDOUT */
```

Comment:
Is this needed? Typically stdout buffered file descriptors flush automatically when their internal buffers are full (or if they are configured for line-buffering, when a newline is encountered). That reduces the number of system calls needed.

I would guess the main fix needed here was replacing `tt_vuart_buf_size()` with `tt_vuart_buf_space()`.

Reply:
OK - it's possibly a side-effect of termios mucking.